### PR TITLE
Update examples to use new APIs and remove `snapp` references

### PR DESCRIPTION
**Pull request by MartinMinkov**
**Description**:
1. Update `snarkyjs` to `0.3.3`
2. Update examples to use new APIs that are exposed by `snarkyjs`
3. Remove any references of `snapp` and replace with `zkApp`

**Tested By**:
Manually testing. Built the new examples and ran them as normal. Compared output with previous iterations to confirm the same behavior.

### CODE REVIEW

**Comment by edwinhere** [comment]
I am getting this error when I try to run your tictactoe.

```
src/index.ts(233,50): error TS2339: Property 'accountCreationFee' does not exist on type 'typeof import("/home/edwin/dev/zk/tictactoe/node_modules/snarkyjs/dist/server/lib/mina")'.
src/index.ts(263,19): error TS2339: Property 'sign' does not exist on type 'TicTacToe'.
src/index.ts(285,19): error TS2339: Property 'sign' does not exist on type 'TicTacToe'.
src/index.ts(307,19): error TS2339: Property 'sign' does not exist on type 'TicTacToe'.
src/index.ts(330,19): error TS2339: Property 'sign' does not exist on type 'TicTacToe'.
src/index.ts(352,19): error TS2339: Property 'sign' does not exist on type 'TicTacToe'.
```

File: examples/sudoku/ts/src/sudoku-zkapp.ts [hunk 1]
```diff
@@ -106,32 +97,32 @@ class SudokuSnapp extends SmartContract {
 const Local = Mina.LocalBlockchain();
 Mina.setActiveInstance(Local);
 const account1 = Local.testAccounts[0].privateKey;
-const account2 = Local.testAccounts[1].privateKey;
 
-const snappPrivkey = PrivateKey.random();
-let snappAddress = snappPrivkey.toPublicKey();
+const zkappKey = PrivateKey.random();
```
Comment (jasongitmail) [inline examples/sudoku/ts/src/sudoku-zkapp.ts:101]: slight preference for `zkAppPrivateKey` b/c of the casing of zkApp and clarity of name for the private key.

File: examples/sudoku/ts/src/sudoku-zkapp.ts [hunk 2]
```diff
@@ -106,32 +97,32 @@ class SudokuSnapp extends SmartContract {
...
+let zkappAddress = zkappKey.toPublicKey();
```
Comment (jasongitmail) [inline examples/sudoku/ts/src/sudoku-zkapp.ts:102]: zkAppAddress

File: examples/sudoku/ts/src/sudoku-zkapp.ts [hunk 3]
```diff
@@ -106,32 +97,32 @@ class SudokuSnapp extends SmartContract {
...
 
 async function deploy(sudoku: number[][]) {
-  let tx = Mina.transaction(account1, () => {
-    const initialBalance = UInt64.fromNumber(1000000);
-    const p = Party.createSigned(account2);
-    p.balance.subInPlace(initialBalance);
-    let snapp = new SudokuSnapp(snappAddress);
+  let tx = await Mina.transaction(account1, () => {
+    Party.fundNewAccount(account1);
+    let zkapp = new SudokuZkapp(zkappAddress);
```
Comment (jasongitmail) [inline examples/sudoku/ts/src/sudoku-zkapp.ts:107]: zkApp

File: examples/sudoku/ts/src/sudoku-zkapp.ts [hunk 4]
```diff
@@ -33,18 +32,10 @@ class Sudoku extends CircuitValue {
   }
 }
 
-class SudokuSnapp extends SmartContract {
+class SudokuZkapp extends SmartContract {
```
Comment (jasongitmail) [inline examples/sudoku/ts/src/sudoku-zkapp.ts:35]: I think we could clean this up a bit by just calling it `Sudoku`

File: examples/sudoku/ts/src/sudoku-zkapp.ts [hunk 5]
```diff
@@ -106,32 +97,32 @@ class SudokuSnapp extends SmartContract {
...
 }
 
 async function submitSolution(sudoku: number[][], solution: number[][]) {
-  let tx = Mina.transaction(account2, () => {
-    let snapp = new SudokuSnapp(snappAddress);
-    snapp.submitSolution(new Sudoku(sudoku), new Sudoku(solution));
-    snapp.self.sign(snappPrivkey);
-    snapp.self.body.incrementNonce = Bool(true);
+  let tx = await Mina.transaction(account1, () => {
+    let zkapp = new SudokuZkapp(zkappAddress);
```
Comment (jasongitmail) [inline examples/sudoku/ts/src/sudoku-zkapp.ts:122]: `zkApp`, `Sudoku`, `zkAppAddress`

**Review by jasongitmail — COMMENTED** [review]
Adding some naming nits, as slightly preferences. Didn't dig into the code or test it, will let @ymekuria

**Review by mitschabaude — APPROVED** [review]

**Pull request merged**